apps/portal/prisma/new-schema.prisma

```diff
@@ -0,0 +1,229 @@
+    userId    String
+    createdAt DateTime @default(now())
+
+    @@unique([userId, roleId])
+}
+
+model User {
+    id            String        @id @default(auto()) @map("_id") @db.ObjectId
+    name          String?
+    email         String?       @unique
+    emailVerified DateTime?
+    image         String?
+    accounts      Account[]
+    sessions      Session[]
+    profile       Profile?
+    roles         RolesOnUser[]
+
+    applications    UserAppJunction[]   @relation("UserLink")
+
+    @@map("users")
+}
+
+model VerificationToken {
+    id         String   @id @default(auto()) @map("_id") @db.ObjectId
+    identifier String
+    token      String   @unique
+    expires    DateTime
+
+    @@unique([identifier, token])
+}
+
+model Profile {
+    id               String    @id @default(auto()) @map("_id") @db.ObjectId
+    firstName        String
+    lastName         String
+    email            String    @unique
+    netid            String    @unique
+    classStanding    String
+    major            String
+    utdStudent       Boolean
+    roles            String[]
+    membershipStatus Boolean
+    membershipTS     DateTime?
+    resume           Boolean
+    resumeTS         DateTime?
+
+    user User @relation(fields: [userId], references: [id])
+
+    userId String             @unique @db.ObjectId
+    events EventReservation[]
+
+    officerId   String             @unique @db.ObjectId
+    officer     Officer            @relation(fields: [officerId], references: [id])
+}
+
+model Event {
+    id          String             @id @default(auto()) @map("_id") @db.ObjectId
+    summary     String
+    description String
+    url         String
+    location    String
+    start       DateTime
+    end         DateTime
+    profiles    EventReservation[]
+
+    @@map("events")
+}
+
+model TypeformApplication {
+    id                  String  @id @default(auto()) @map("_id") @db.ObjectId
+    active              Boolean
+    description         String
+    endpoint            String
+    externalResourceUrl String
+    typeformId          String
+    typeformName        String
+
+    @@map("typeform_applications")
+}
+
+model Application {
+    id                  String  @id @default(auto()) @map("_id") @db.ObjectId
+    createdAt           DateTime
+    expireDate          DateTime
+    type                String?
+    role                String?
+
+    users               UserAppJunction[]   @relation("AppLink")
+    departments         UserAppJunction[]   @relation("AppLink")
+
+    @@map("applications")
+}
+
+model Department {
+    id                  String  @id @default(auto()) @map("_id") @db.ObjectId
+    deptName            String
+
+    officers            Officer[]
+    applications        UserAppJunction[]   @relation("DeptLink")
+
+    @@map("departments")
+}
+
+model Officer {
+    id                  String  @id @default(auto()) @map("_id") @db.ObjectId
+    deptId              String
```

Comment:
Shouldn't this be `String[]`, since an officer can be a part of multiple division?

Reply:
That actually needs to be removed by me - since the departments variable which is an array of Department takes care of that